src/Illuminate/Validation/Rules/Email.php

```diff
@@ -209,33 +210,35 @@ public function passes($attribute, $value)
     /**
      * Build the array of underlying validation rules based on the current state.
      *
+     * @param  string  $attribute
+     * @param  mixed  $value
      * @return array
      */
-    protected function buildValidationRules()
+    public function buildValidationRules($attribute, $value)
     {
         $rules = [];
 
-        if ($this->rfcCompliant) {
+        if (value($this->rfcCompliant, $attribute, $value)) {
             $rules[] = 'rfc';
         }
 
-        if ($this->strictRfcCompliant) {
+        if (value($this->strictRfcCompliant, $attribute, $value)) {
             $rules[] = 'strict';
         }
```

Comment:
Furthermore, it would be interesting, yet not easily implemented with the current implementation, if the method didn't only return whether to apply a specific rule but also if the more general rule should be applied with a flag (in this case `strict`). Implementing this by making the first param of the setter function `bool|\Closure(): bool` as well, would be one option, only passing in `bool|Closure(): RfcComplianceEnum|null $condition` another.

Our hypothetical `RfcComplianceEnum` would look something like this:
```php
enum RfcComplianceEnum
{
    case Loose; // or Basic
    case Strict;
}
```